### PR TITLE
*: postpone drop bundle for truncate/drop partition/drop schema to gc

### LINKED ISSUE #29827: postpone some bundle remove operation to gc

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #29827

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by AilinKid — APPROVED** [review]
LGTM

File: ddl/placement_policy_test.go [hunk 1]
```diff
@@ -974,6 +990,91 @@ func (s *testDBSuite6) TestDatabasePlacement(c *C) {
...
+
+	tk.MustExec("drop database db2")
+
+	bundles, err := infosync.GetAllRuleBundles(context.TODO())
+	c.Assert(err, IsNil)
+	c.Assert(len(bundles), Equals, 4)
+
+	gcWorker, err := gcworker.NewMockGCWorker(s.store)
+	c.Assert(err, IsNil)
+	c.Assert(gcWorker.DeleteRanges(context.TODO(), math.MaxInt64), IsNil)
```
Comment (xhebox) [inline ddl/placement_policy_test.go:1028]: I remembered that gc will not handle `dbInfo.ID`, does that mean `policy of db` will not be deleted?

File: ddl/placement_policy_test.go [hunk 2]
```diff
@@ -974,6 +990,91 @@ func (s *testDBSuite6) TestDatabasePlacement(c *C) {
 	))
 }
 
+func (s *testDBSuite6) TestDropDatabaseGCPlacement(c *C) {
+	clearAllBundles(c)
```
Comment (xhebox) [inline ddl/placement_policy_test.go:994]: That our tests will call the global `infoscyner` and `failpoints`, is it safe to run in the parallel suite?

**Review by xhebox — COMMENTED** [review]
While it is possible to do so, I want to learn that why do you want to postpone the deletion to gc?

File: ddl/placement_policy_test.go (hunk 2 above)
Comment (lcwangchao) [reply ddl/placement_policy_test.go:994]: I think it's safe on condition that all placement sql are executed in serial

File: ddl/placement_policy_test.go (hunk 1 above)
Comment (lcwangchao) [reply ddl/placement_policy_test.go:1028]: I'm completing the missing cases  for https://github.com/pingcap/tidb/pull/28687 . But I think the reasons is:

1. Deleting rules when dropping may cause the reschedule of PD. I'm not sure if it'll introduce some extra cost.
2. Users may still expect their data fits the placement rule, even it is dropped. For example: compliance
3. Users can still access deleted data even if it is deleted like stale read. At this time they still want the data fits the placement rule for some reason like isolation.

**Comment by lcwangchao** [comment]
> While it is possible to do so, I want to learn that why do you want to postpone the deletion to gc?

I'm completing the missing cases  for https://github.com/pingcap/tidb/pull/28687 . But I think the reasons is:

1. Deleting rules when dropping may cause the reschedule of PD. I'm not sure if it'll introduce some extra cost.
2. Users may still expect their data fits the placement rule, even it is dropped. For example: compliance
3. Users can still access deleted data even if it is deleted like stale read. At this time they still want the data fits the placement rule for some reason like isolation.

**Review by xhebox — APPROVED** [review]

**Pull request merged**